src/main/java/org/sagebionetworks/bridge/udd/synapse/SynapsePackager.java

```diff
@@ -219,16 +263,69 @@ public PresignedUrlInfo packageSynapseData(Map<String, UploadSchema> synapseToSc
 
         // write errors into an error log file for the user
         if (!errorList.isEmpty()) {
-            String errorLog = LINE_JOINER.join(errorList);
-            File errorLogFile = fileHelper.newFile(tmpDir, ERROR_LOG_FILE_NAME);
+            File errorLogFile = writeErrorLog(errorList, ERROR_LOG_FILE_NAME, tmpDir);
+            allFileList.add(errorLogFile);
+        }
+        return allFileList;
+    }
 
-            try (Writer errorLogFileWriter = fileHelper.getWriter(errorLogFile)) {
-                errorLogFileWriter.write(errorLog);
+    /**
+     * Waits for the survey metadata async tasks. Returns a list of the downloaded metadata files. The list also
+     * includes an error log, if there are any errors.
+     *
+     * @param tmpDir
+     *         temp directory files should be downloaded to and error log should be written to
+     * @param futureList
+     *         list of Futures for async tasks that should be waited on
+     * @return list of all files downloaded, plus error log
+     * @throws IOException
+     *         if writing the error log fails
+     */
+    private List<File> waitForAsyncSurveyTasks(File tmpDir, List<Future<File>> futureList) throws IOException {
+        // join on threads until they're all done
+        List<File> fileList = new ArrayList<>();
+        List<String> errorList = new ArrayList<>();
+        for (Future<File> oneFuture : futureList) {
+            try {
+                File file = oneFuture.get();
+                fileList.add(file);
+            } catch (ExecutionException | InterruptedException ex) {
+                String errorMsg = "Error downloading survey: " + ex.getMessage();
+                LOG.error(errorMsg, ex);
+                errorList.add(errorMsg);
```

Comment:
From my understanding, the only way to get an InterruptedException is if we explicitly interrupt the thread, which we never do. Regardless, I still prefer to err on the side of not crashing the entire request when an InterruptedException gets thrown, so I don't think propagating the InterruptedException is the right thing to do.

That said, I think calling Thread.currentThread().interrupt() is the right thing to do, since it'll properly track the interrupted status (even though no one is actually reading it).